### ISSUE TITLE
b:syntastic_skip_checks is being set against my will, stopping all checks in python files

### ISSUE THREAD

**Issue by oryband** [scrooloose/syntastic#1603]
The `b:syntastic_skip_checks` is being set to `1` when randomly working on python files. I don't know what is causing this. The only things I can remember using is NerdTree to open more python files, and Ctrl-P to navigate and close them.

Does anybody have any idea what might be causing this?

Here is the tail of `:messages`:
```vim
"python1.py" 203L, 6618C [w]
"python2.py" 62L, 1884C
"python3.py" [+] 205 lines --87%--
" some more lines like this
Syntastic version: 3.7.0-20 (Vim 704, Linux)
Info for filetype: python
Global mode: active
Filetype python is active
The current file will be checked automatically
Available checkers: flake8 pep8 pyflakes python
Currently enabled checker: flake8
The current file will not be checked (b:syntastic_skip_checks set)
```

**Comment by lcd047** [comment]
Syntastic never sets this variable, it only reads it.

**Comment by oryband** [comment]
@lcd047 grepping through my vundle plugin library, I didn't find any code line that sets it:
```bash
.vim/bundle [master] λ ag "b:syntastic_skip_checks"
syntastic/doc/syntastic.txt
251:|'b:syntastic_skip_checks'|.
614:                                                   *'b:syntastic_skip_checks'*
616:Only the local form |'b:syntastic_skip_checks'| is used. When set to a true
618:    let b:syntastic_skip_checks = 1

syntastic/plugin/syntastic.vim
291:    let b:syntastic_skip_checks = get(b:, 'syntastic_skip_checks', 0) || !syntastic#util#var('check_on_wq')
647:            call add(why, 'b:syntastic_skip_checks set')
```
I understand that this variable is not set by syntastic, but do you have any idea what might cause this? At least for the sake of documentation.

**Comment by lcd047** [comment]
Looking at your `vimrc` or one of the files sourced from it would be a good place to start.  If you can't find it there, can you consistently reproduce the problem?  Is the variable actually set when syntastic stops working?

**Comment by oryband** [comment]
@lcd047 i tried reproducing it, but this happens randomly, even without trigger NerdTree. Just working on a single python file and saving it (triggering `:SyntasticCheck`), and then after a while it randomly stops checking since that variable gets set.

And yes, setting `let b:syntastic_skip_checks set = 0` and `:w` and then syntastic starts checking the file again.

**Comment by oryband** [comment]
I'm also not sourcing anything there besides the plugins i already grepped on.

**Comment by lcd047** [comment]
When syntastic stops working, is the variable actually set?  That is, you can `:echo b:syntastic_skip_checks set` and get back a value (presumably 1), as opposed to an error message?

**Comment by oryband** [comment]
@lcd047 exactly, yes.

**Comment by lcd047** [comment]
Does any of your other plugins set a `QuitPre` autocommand?

**Comment by oryband** [comment]
@lcd047 Doesn't look like it:
```vim
.vim/bundle [master] λ ag QuitPre
syntastic/plugin/syntastic.vim
248:if exists('##QuitPre')
249:    " QuitPre was added in Vim 7.3.544
251:        autocmd QuitPre * call s:QuitPreHook()
288:function! s:QuitPreHook() abort " {{{2
290:        \ 'autocmd: QuitPre, buffer ' . bufnr('') . ' = ' . string(bufname(str2nr(bufnr('')))))
```

**Comment by oryband** [comment]
Maybe this will help?
```vim
.vim/bundle [master] λ vim --version
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Sep 28 2015 08:50:43)
Included patches: 1-884
Compiled by Arch Linux
Huge version with GTK2 GUI.  Features included (+) or not (-):
+acl             +farsi           +mouse_netterm   +syntax
+arabic          +file_in_path    +mouse_sgr       +tag_binary
+autocmd         +find_in_path    -mouse_sysmouse  +tag_old_static
+balloon_eval    +float           +mouse_urxvt     -tag_any_white
+browse          +folding         +mouse_xterm     -tcl
++builtin_terms  -footer          +multi_byte      +terminfo
+byte_offset     +fork()          +multi_lang      +termresponse
+cindent         +gettext         -mzscheme        +textobjects
+clientserver    -hangul_input    +netbeans_intg   +title
+clipboard       +iconv           +path_extra      +toolbar
+cmdline_compl   +insert_expand   +perl            +user_commands
+cmdline_hist    +jumplist        +persistent_undo +vertsplit
+cmdline_info    +keymap          +postscript      +virtualedit
+comments        +langmap         +printer         +visual
+conceal         +libcall         +profile         +visualextra
+cryptv          +linebreak       +python          +viminfo
+cscope          +lispindent      -python3         +vreplace
+cursorbind      +listcmds        +quickfix        +wildignore
+cursorshape     +localmap        +reltime         +wildmenu
+dialog_con_gui  +lua             +rightleft       +windows
+diff            +menu            +ruby            +writebackup
+digraphs        +mksession       +scrollbind      +X11
+dnd             +modify_fname    +signs           -xfontset
-ebcdic          +mouse           +smartindent     +xim
+emacs_tags      +mouseshape      -sniff           +xsmp_interact
+eval            +mouse_dec       +startuptime     +xterm_clipboard
+ex_extra        +mouse_gpm       +statusline      -xterm_save
+extra_search    -mouse_jsbterm   -sun_workshop    -xpm
   system vimrc file: "/etc/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "/etc/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-2.0 -I/usr/lib/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/libdrm -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include  -D_FORTIFY_SOURCE=2  -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong --param=ssp-buffer-size=4 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: gcc   -L. -Wl,-O1,--sort-common,--as-needed,-z,relro -fstack-protector -rdynamic -Wl,-export-dynamic -Wl,-E -Wl,-rpath,/usr/lib/perl5/core_perl/CORE  -Wl,-O1,--sort-common,--as-needed,-z,relro -L/usr/local/lib -Wl,--as-needed -o vim   -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lfontconfig -lfreetype  -lSM -lICE -lXt -lX11 -lXdmcp -lSM -lICE  -lm -lncurses -lelf -lnsl   -lacl -lattr -lgpm -ldl  -L/usr/lib -llua -Wl,-E -Wl,-rpath,/usr/lib/perl5/core_perl/CORE -Wl,-O1,--sort-common,--as-needed,-z,relro -fstack-protector-strong -L/usr/local/lib  -L/usr/lib/perl5/core_perl/CORE -lperl -lpthread -lnsl -ldl -lm -lcrypt -lutil -lc -L/usr/lib/python2.7/config -lpython2.7 -lpthread -ldl -lutil -lm -Xlinker -export-dynamic   -lruby -lpthread -lgmp -ldl -lcrypt -lm  -L/usr/lib
```

**Comment by lcd047** [comment]
`QuitPre` is case-insensitive.

**Comment by oryband** [comment]
@lcd047 same results:
```vim
.vim/bundle [master] λ ag quitpresyntastic/plugin/syntastic.vim
248:if exists('##QuitPre')
249:    " QuitPre was added in Vim 7.3.544
251:        autocmd QuitPre * call s:QuitPreHook()
288:function! s:QuitPreHook() abort " {{{2
290:        \ 'autocmd: QuitPre, buffer ' . bufnr('') . ' = ' . string(bufname(str2nr(bufnr('')))))
```

**Comment by lcd047** [comment]
Please set `g:syntastic_debug` to 8 in your `vimrc`, and run Vim normally until syntastic stops working.  Then run `:mes` and post the output.

**Comment by lcd047** [comment]
Ok, pilot error here.  This should be fixed in 77e20e2.

**Comment by oryband** [comment]
thanks! didn't happen since i updated to latest version. will ask to reopen if this occurs again.
